### ISSUE TITLE
Default to 2 whitespaces instead of 4 for `normalise`?

### ISSUE THREAD

**Issue by Jeehut** [SDGGiesbrecht/Workspace#598]
### Question

Just to try this tool out, I opened one of my projects and ran `workspace normalise` which resulted in all my filed to be converted from a 4-whitespaces based indentation to 2-whitespaces based indentation. I wonder, how is 2-whitespaces indentation considered best practice for Swift projects? Shouldn't we default to 4-whitespaces based indentation? Or did it even convert my 4-whitespaces to tabs (which would be an even worse default IMHO)?

### Documentation Suggestion

I'd at least expect to find a way to configure this behavior, which I can't find documented anywhere. As a matter of fact, I didn't even find any documentation on the behavior overall. The documentation for `normalise` only states:

> Normalises the project’s files by removing trailing whitespace, applying Unix newlines, performing canonical decomposition and formatting Swift files.

The reindentation isn't mentioned anywhere, or is that part of `formatting Swift files`? Then I wonder, how is that formatting done, using SwiftFormat? What configuration is applied to SwiftFormat, the default one? Does it really use 2-whitespaces indentation?

**Comment by SDGGiesbrecht** [comment]
The default of 2‐space indentation is inherited from [swift‐format](https://github.com/apple/swift-format), which is part of the Swift open source project. And yes, “formatting Swift files” was intended to mean passing each file off to swift‐format to do whatever formatting the Swift project decides should be standard.

However, at least for a basic `workspace refresh`, I the default is supposed to be a `nil` swift‐format configuration, which would exclude swift‐format entirely (precisely because of this invasiveness). Only if `optIntoAllTasks()` were applied would it end up with swift‐format’s standard configuration.

Calling from the command line normally overrides the configuration. For example, having file header management turned off (so that it doesn’t happen during `workspace refresh`), but explicitly doing `workspace refresh file‐headers` would still rewrite the file headers. Something in that plumbing might be activating the more aggressive swift‐format configuration. Does it do the same re‐indentation on a fresh project with both `workspace refresh` and `workspace normalize`, or are the results different?

**Comment by SDGGiesbrecht** [comment]
Okay, I did some experimenting and looked back at the history. I had it slightly mixed up in my memory.

- The default doesn’t normalize *at all*.
- Applying `optIntoAllTasks()` to the configuration will turn normalization on permanently.
- Explicitly running `workspace normalize` overrides the configuration and turns it on for the duration of the command.
- (Nothing makes the `swift-format` configuration change between modes.)

This is both the observed and intended behaviour.

***

As far as customizing it to use 4‐space indentation, the configuration entry would be this:

```swift
configuration.proofreading.swiftFormatConfiguration?.indentation = .spaces(4)
```

The value of the `swiftFormatConfiguration` property is the Swift type vended by swift‐format’s [`SwiftFormatConfiguration` library](https://github.com/apple/swift-format/blob/8df2cb04f2aa41089cd14379552cb97bcc89133a/Package.swift#L27), which is documented to some degree [here](https://github.com/apple/swift-format/blob/main/Documentation/Configuration.md).

***

Since everything is doing what it is supposed to, I am reclassifying this issue as a documentation gap.

**Comment by SDGGiesbrecht** [comment]
Would the additions from #606 have pre‐empted your question, or is there more you think I should add?

```shell
$ workspace normalise help
```
> `workspace normalise` normalises the project’s files by removing trailing whitespace, applying Unix newlines, performing canonical decomposition and formatting Swift files.
>
> The general formatting is done by applying Swift’s official formatter, known as swift‐format. More information about it is available at ‹https://github.com/apple/swift-format›.
>
> Inside a Workspace configuration, the configuration for swift‐format is found under ‘proofreading.swiftFormatConfiguration’ and it uses the Swift API from its own SwiftFormatConfiguration module. Setting the entire property to nil will completely disable swift‐format.

**Comment by Jeehut** [comment]
@SDGGiesbrecht Thank you for asking: I think, if the same text is also shown in the docs online, then it will surely help.

**Comment by SDGGiesbrecht** [comment]
The generated HTML documentation will inherit the change automatically as soon as the PR is merged.

Unfortunately, the best documentation swift‐format seems to provide at the moment is the comments in its source code. Do you think this link would be better?: https://github.com/apple/swift-format/blob/0.50300.0/Sources/SwiftFormatConfiguration/Configuration.swift#L20

**Comment by SDGGiesbrecht** [comment]
I added that second link and merged the PR.

**Comment by SDGGiesbrecht** [comment]
The improvements to the `help` subcommand are included in 0.35.3.
